### PR TITLE
UNSTOPPABLE flag is now checked every time something tries to pass something instead of just once in the Move() proc

### LINKED ISSUE #41980: Reflector boxes/emitters are behaving wierdly.
Issue reported from  Round ID: 98705 (/tg/Station Terry [EU] [100% LAG FREE])

Reporting client version: 512

See the gif:
https://i.imgur.com/LNtvsyg.gif

### PR DESCRIPTION
**Pull request by silicons**
This means that if it's changed during a move it'll be respected instead of ignored.
Fixes #41980 
@optimumtact this might be performance nightmare so uhhh

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Can I ask how this compares with something we already have in place aka pass flags?

**Comment by kevinz000** [comment]
@ExcessiveUseOfCobblestone unstoppable flag allows something to move where-ever it is directed to ignoring but still triggering all Bump()s and Cross()es
Passflags are basically noclipping across certain things without bumping, last I checked. And passflags aren't all-inclusive while unstoppable is

**Comment by YakumoChen** [comment]
150 hours of kevin

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -221,22 +220,22 @@
...
+				if(CHECK_BITFIELD(mover.movement_type, UNSTOPPABLE))
+					mover.Bump(thing)
 					continue
 				else
 					if(!firstbump || ((thing.layer > firstbump.layer || thing.flags_1 & ON_BORDER_1) && !(firstbump.flags_1 & ON_BORDER_1)))
 						firstbump = thing
 	if(firstbump)
-		mover?.Bump(firstbump)
-		return unstoppable
+		if(!QDELETED(mover))
```
[the remark this answers is not in the document]
Comment (optimumtact) [reply code/game/turfs/turf.dm]: @kevinz000 why does it take you 3days to respond to a comment

[the remark this answers is not in the document]
Comment (kevinz000) [reply code/game/turfs/turf.dm]: uhhhhhhh

**Pull request merged**